### ISSUE TITLE
Godot crashes while typing access to a field of exported variable in the gdscript editor

### ISSUE THREAD

**Issue by OliPerraul** [godotengine/godot#20589]
**Godot version:**
![image](https://user-images.githubusercontent.com/24371247/43374009-743645bc-937b-11e8-91a5-0110f52bc6a2.png)

**Issue description:**
Godot crashes after when accessing field of exported variable inside instanced scene. Possibly related to #19264
![image](https://user-images.githubusercontent.com/24371247/43374177-515d0746-937c-11e8-84ea-31794b5efa6c.png)

**Steps to reproduce:**
1. write
```
export(NodePath) onready var _tree_node = get_node(_tree_node)
```
2. in `func _ready()` begin writing `_tree_node.` with `.` operator. 
3. Notice godot crashes from the editor

**Comment by OliPerraul** [comment]
Doing this fixes the issue

```

export(NodePath) var _root_node
export(NodePath) var _tree_node


export(int) var _preview_item_max = 10
export(float) var _preview_opacity_max = 1.0
var opacity_step = _preview_opacity_max/_preview_item_max

var _drag_preview
var _mouse_pos



func _ready():
	_root_node = get_node(_root_node)
	_tree_node = get_node(_tree_node) 

```

**Comment by T4g1** [comment]
Some quick context from gdb (tested on a custom branch though):
```
#0  0x00005555591ef292 in Variant::reference (this=<error reading variable: Cannot access memory at address 0x7fffff7fed68>, p_variant=<error reading variable: Cannot access memory at address 0x7fffff7fed60>) at core/variant.cpp:905
#1  0x00005555591faa80 in Variant::operator= (this=0x7fffff801420, p_variant=...) at core/variant.cpp:2504
#2  0x000055555649ebcb in _guess_expression_type (p_context=..., p_expression=0x5555600a5b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:650
#3  0x00005555564a03fc in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:738
#4  0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#5  0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#6  0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#7  0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#8  0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#9  0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#10 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#11 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#12 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#13 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#14 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#15 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#16 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#17 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#18 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#19 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#20 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#21 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#22 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#23 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#24 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#25 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#26 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#27 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#28 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#29 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#30 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#31 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#32 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#33 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#34 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#35 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#36 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
#37 0x00005555564a8e83 in _guess_identifier_type (p_context=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1291
#38 0x000055555649ec1c in _guess_expression_type (p_context=..., p_expression=0x5555602d0b20, r_type=...) at modules/gdscript/gdscript_editor.cpp:656
#39 0x00005555564a0b2a in _guess_expression_type (p_context=..., p_expression=0x5555602d0a40, r_type=...) at modules/gdscript/gdscript_editor.cpp:779
#40 0x00005555564aa260 in _guess_identifier_type_from_base (p_context=..., p_base=..., p_identifier=..., r_type=...) at modules/gdscript/gdscript_editor.cpp:1373
```

**Comment by KoBeWi** [comment]
Any more reproduction steps? Doing what is described in OP seems to work in 3.1 stable.

**Comment by akien-mga** [comment]
Assuming fixed in 3.1. Please comment if you can still reproduce it.
